lib/main.dart

```diff
@@ -18,7 +58,26 @@ class HomePage extends StatefulWidget {
 class _HomePageState extends State<HomePage> {
   final GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
   int currentIndex = 0;
-  var title = ['Chats','Contacts','Reels'];
+  var title = ['Chats', 'Contacts', 'Reels'];
+  List<User> users = [];
+
+  @override
+  void initState() {
+    super.initState();
+    Future.microtask(() async {
+      // listen realtime
+      FirebaseFirestore.instance
+          .collection("users")
+          .snapshots()
+          .listen((event) {
+        final users = event.docs.map((e) => User.fromMap(e.data())).toList();
+        print(users);
+        setState(() {
+          this.users = users;
+        });
+      });
+    });
```

Comment:
lắng nghe toàn bộ data trong table 'users', khi có data nào change thì nó sẽ gọi lại hàm này. Sau đó gọi hàm setState để update lại list `users` và UI sẽ update theo.